pytorch_lightning/trainer/trainer.py

```diff
@@ -1441,58 +1447,106 @@ def _call_teardown_hook(self) -> None:
-            # The `Accelerator` mostly calls the `TrainingTypePlugin` but some of those calls are deprecated.
-            # The following logic selectively chooses which hooks are called on each object.
-            # In the case of `setup` and `teardown`, the hooks on the `LightningModule` should not call the hooks of the
-            # same name in these objects as they are meant to be managed outside of the `LightningModule` lifecycle.
-            # All of this should be fixed by #8506
-
-            # call the accelerator hook
-            if hook_name in ("on_train_start",) and hasattr(self.accelerator, hook_name):
-                accelerator_hook = getattr(self.accelerator, hook_name)
-                accelerator_output = accelerator_hook(*args, **kwargs)
-                # Rely on the accelerator output if lightningModule hook returns nothing
-                # Required for cases such as DataParallel where we reduce the output for the user
-                # todo: move this data parallel logic into the data parallel plugin
-                output = accelerator_output if output is None else output
-
-            # call the ttp hook
-            if hook_name not in ("setup", "teardown", "on_train_start") and hasattr(
-                self.training_type_plugin, hook_name
-            ):
-                ttp_hook = getattr(self.training_type_plugin, hook_name)
-                ttp_output = ttp_hook(*args, **kwargs)
-                output = ttp_output if output is None else output
+        # TODO: remove if statement in v1.7
+        if hook_name in ("on_train_batch_start", "on_train_batch_end"):
+            fn = getattr(self, hook_name)
+            if callable(fn):
+                with self.profiler.profile(hook_name):
+                    output = fn(*args, **kwargs)
+        else:
+            for callback in self.callbacks:
+                fn = getattr(callback, hook_name)
+                if callable(fn):
+                    with self.profiler.profile(hook_name):
+                        output = fn(self, self.lightning_module, *args, **kwargs)
 
         if pl_module:
             # restore current_fx when nested context
             pl_module._current_fx_name = prev_fx_name
 
         return output
 
+    # TODO: rename to _call_strategy_hook and eventually no longer need this
+    def _call_ttp_hook(
+        self,
+        hook_name: str,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        fn = getattr(self.training_type_plugin, hook_name)
+        if not callable(fn):
+            return None
+
+        with self.profiler.profile(hook_name):
+            output = fn(*args, **kwargs)
+
+        return output
+
+    # TODO: eventually no longer need this
+    def _call_accelerator_hook(
+        self,
+        hook_name: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Optional[Any]:
+        fn = getattr(self.accelerator, hook_name)
+        if not callable(fn):
+            return None
+
```

Comment:
We discussed not setting the `current_fx_name` for accelerator but since we do it manually in all places that call this method, might as well put it inside.

Reply:
Yeah, I ended up having to change it because some tests were failing. Also at least this way we don't use the LM's trainer reference in TTP.py

Reply:
I have some mind boggling problems in #10890. One question, why do these new call_xyz methods not reset the current_fx_name to its previous value? Was this forgotten or a mistake?

Reply:
After discussion with @carmocca I learned that only LM hooks and callback hooks need to set current_fx_name, so I only included that in the `_call_LM_hook` and `_call_callback_hooks`, and not for `_call_ttp_hook` and `_call_accelerator_hook`. However, before this PR, all the places where we called the accelerator hook had a line like this:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3d6262b7a91215e72019d720e742e6261e1636dc/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py#L220

so I included `self.lightning_module._current_fx_name = hook_name` in `_call_accelerator_hook` and got rid of the individual lines like the above